### PR TITLE
Add doc for remote relations worker

### PR DESCRIPTION
**Pull request by wallyworld**
A simple addition of doc for the remote relations worker.

### CODE REVIEW

File: worker/remoterelations/doc.go
Comment (barrettj12) [inline worker/remoterelations/doc.go:1]: I think we generally stick to a 80-char line length, so let's observe that here.

File: worker/remoterelations/doc.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package remoterelations contains workers which manage the operation of cross model relations.
```
Comment (barrettj12) [inline worker/remoterelations/doc.go:4]: Can we enumerate the multiple workers and describe what each one does?

File: worker/remoterelations/doc.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+// The workers run on both the consuming and offering models.
```
Comment (barrettj12) [inline worker/remoterelations/doc.go:5]: Which workers run where? Is it one worker on consuming, one worker on offering, or both workers on both sides?

File: worker/remoterelations/doc.go [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+// The consuming side pushes relation updates from the consumer application to the model containing
+// the offer. It also polls the offered application to record relation changes from the offer into
```
Comment (barrettj12) [inline worker/remoterelations/doc.go:7]: "to the consuming model" makes more sense to me

**Review by barrettj12 — CHANGES_REQUESTED** [review]

File: worker/remoterelations/doc.go (hunk 1 above)
Comment (tmihoc) [reply worker/remoterelations/doc.go:4]: Also, maybe better to say: defines* workers that...

**Review by tmihoc — CHANGES_REQUESTED** [review]

File: worker/remoterelations/doc.go
Comment (wallyworld) [reply worker/remoterelations/doc.go:1]: This isn't consistently used anywhere across the codebase. I think we tend to stick to what reads well.

File: worker/remoterelations/doc.go (hunk 1 above)
Comment (wallyworld) [reply worker/remoterelations/doc.go:4]: Workers don't have names. We can attempt to describe them by the struct the implements them.

File: worker/remoterelations/doc.go (hunk 2 above)
Comment (wallyworld) [reply worker/remoterelations/doc.go:5]: Added an extra bit of clarification

**Review by tmihoc — APPROVED** [review]
LGTM, thanks!

**Review by tmihoc — CHANGES_REQUESTED** [review]
Actually found some typos and punctuation and spelling issues. Suggested update below:

> Package `remoterelations` defines workers which manage the operation of cross model relations.

- `Worker`: Top level worker. Watches SaaS applications/proxies and creates a worker for each.
- `remoteApplicationWorker`: Manages operations for a consumer or offer proxy. Consumes and publishes relation data and status changes.
- `remoteRelationsWorker`: Runs on the consuming model to manage relations to the offer.
- `relationUnitsWorker`: Runs on the consuming model to receive and publish changes to each relation unit data bag.

The consuming side pushes relation updates from the consumer application to the model containing
the offer. It also polls the offered application to record relation changes from the offer into
the consuming model.

**Comment by wallyworld** [comment]
Suggested changes made except for the quotes around "Package remoterelation" since the docstring just needs to be `Package foo ...`.

**Review by tmihoc — APPROVED** [review]
LGTM, thanks!

**Pull request merged**